lib/mysql_connection.cpp

```diff
@@ -210,50 +283,9 @@ MySQL_Connection::MySQL_Connection() {
 	options.no_backslash_escapes=false;
 	options.init_connect=NULL;
 	options.init_connect_sent=false;
-	options.character_set_results = NULL;
```

Comment:
I think the initialization of all this variables have been moved to the _for loop_ in the line 273, but I'm not able to confirm if every variable that was before NULL initialized here is being initialized to NULL inside that loop, since I can see that the enum "variable_name" doesn't hold all the values specified here, maybe it's simply that it isn't relevant and more variables that needed were NULL initialized here before.

Reply:
The `tx_isolation` is a deprecating mysql variable and is handled by internal state of the `isolation level` variable. https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_tx_isolation

Other mysql variables may have additional variable which represents sent flag, which is not used with new implementation. 

Three new variables were added `character_set_client`, `character_set_database`, `character_set_connection`.

The automated test `./test/tap/tests/set_testing-t.cpp` tests all variables in new implementation.

Reply:
That clarifies it, thanks for the info!